controllers/factory/vmagent.go

```diff
@@ -622,6 +642,50 @@ func CreateOrUpdateRelabelConfigsAssets(ctx context.Context, cr *victoriametrics
 	return rclient.Update(ctx, assestsCM)
 }
 
+// buildVMAgentStreamAggrConfig combines all possible stream aggregation configs and adding it to the configmap.
+func buildVMAgentStreamAggrConfig(cr *victoriametricsv1beta1.VMAgent) (*corev1.ConfigMap, error) {
+	cfgCM := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:       cr.Namespace,
+			Name:            cr.StreamAggrConfigName(),
+			Labels:          cr.AllLabels(),
+			Annotations:     cr.AnnotationsFiltered(),
+			OwnerReferences: cr.AsOwner(),
+		},
+		Data: make(map[string]string),
+	}
+	for i, rw := range cr.Spec.RemoteWrite {
+		if !rw.HasStreamAggr() {
+			continue
+		}
+		data, err := yaml.Marshal(rw.StreamAggrConfig.Rules)
+		if err != nil {
+			return nil, fmt.Errorf("cannot serialize StreamAggrConfig rules as yaml: %w", err)
```

Comment:
Can you add idx or remote write url to error context?

Reply:
done